src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,113 @@
+package core.basesyntax.service;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class RegistrationServiceImplTest {
+    private static RegistrationServiceImpl rsi;
+    private User actual = new User();
+
+    @BeforeAll
+    static void beforeAll() {
+        User u = new User();
+        u.setAge(20);
+        u.setLogin("vasyasho");
+        u.setPassword("123456");
+        Storage.people.add(u);
+        User u1 = new User();
+        u1.setAge(21);
+        u1.setLogin("vasyahuha");
+        u1.setPassword("123456789");
+        Storage.people.add(u1);
+        rsi = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp () {
+        actual.setPassword("123456");
+        actual.setLogin("fffffffff");
+        actual.setAge(18);
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        actual.setLogin(null);
+        assertThrows(InvalidUserException.class, () -> {
+            rsi.register(actual);
+        });
+    }
+
+    @Test
+    void register_emptyLogin_notOk() {
+        actual.setLogin("");
+        assertThrows(InvalidUserException.class, () -> {
+            rsi.register(actual);
+        });
+    }
+
+    @Test
+    void register_shortLogin_notOk() {
+        actual.setLogin("vasya");
+        assertThrows(InvalidUserException.class, () -> {
+            rsi.register(actual);
+        });
+    }
+
+    @Test
+    void register_nullAge_notOk() {
+        actual.setAge(null);

```

Comment:
The `User` class likely does not support `null` for primitive types like `int` for age. Consider using a wrapper class like `Integer` if you want to test for `null` values.